token/program-2022/src/processor.rs

```diff
@@ -471,21 +471,32 @@ impl Processor {
     pub fn process_revoke(program_id: &Pubkey, accounts: &[AccountInfo]) -> ProgramResult {
         let account_info_iter = &mut accounts.iter();
         let source_account_info = next_account_info(account_info_iter)?;
-        let owner_info = next_account_info(account_info_iter)?;
-        let owner_info_data_len = owner_info.data_len();
+        let authority_info = next_account_info(account_info_iter)?;
+        let authority_info_data_len = authority_info.data_len();
 
         let mut source_account = Account::unpack(&source_account_info.data.borrow())?;
         if source_account.is_frozen() {
             return Err(TokenError::AccountFrozen.into());
         }
 
-        Self::validate_owner(
-            program_id,
-            &source_account.owner,
-            owner_info,
-            owner_info_data_len,
-            account_info_iter.as_slice(),
-        )?;
+        match source_account.delegate {
```

Comment:
Does the borrow checker 🤯 on this form?
```
Self::validate_owner(
    program_id,
    match source_account.delegate {
      COption::Some(ref delegate) if cmp_pubkeys(authority_info.key, delegate) => delegate,
      _ => &source_account.owner,
    },
    authority_info,
    authority_info_data_len,
    account_info_iter.as_slice(),
)?;
```

In my brain this makes it more clear how the check differs across the two code paths.  But I'm just nitting, rest looks great here

Reply:
It's totally happy with it, nice suggestion!